tests/test_ops.py

```diff
@@ -427,6 +427,166 @@ def test_overlap():
     )
     assert len(b) == 3
 
+    ### test overlap with point and segment data
+    df1 = pd.DataFrame(
+        [
+            ['chr1', 1, 1]
+        ],
+        columns=['chrom','start','end']
+    ).astype({"start": pd.Int64Dtype(), "end": pd.Int64Dtype()})
+
+    df2 = pd.DataFrame(
+        [
+            ['chr1', 1, 2]
+        ],
+        columns=['chrom','start','end']
+    ).astype({"start": pd.Int64Dtype(), "end": pd.Int64Dtype()})
+
+    b = bioframe.overlap(
+        df1,
+        df2,
+        on=None,
+        how="left",
+        return_index=True,
+        return_input=False,
+    )
+    assert np.sum(pd.isna(b["index_"].values)) == 0
```

Comment:
it would be cleaner code if [Issue](https://github.com/open2c/bioframe/issues/188) is resolved first and then code is refactored to incorporate df_result  with complete dataframe in assert function

Reply:
agreed-- but we can modify this test to be more clear what it's goals are even before cleaning up the tests more generally as proposed in #188

e.g. here it would be more clear to just have the result dataframe be specified, and then testing for equivalence of the overlap with the desired result, rather than this current test of no nans in the output

Reply:
i've updated test cases.